domain/src/test/java/org/oppia/android/domain/platformparameter/PlatformParameterSingletonTest.kt

```diff
@@ -0,0 +1,163 @@
+package org.oppia.android.domain.platformparameter
+
+import android.app.Application
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+import org.oppia.android.app.model.PlatformParameter as ParameterValue
+
+private const val STRING_PLATFORM_PARAMETER_NAME = "string_platform_parameter_name"
+private const val STRING_PLATFORM_PARAMETER_VALUE = "string_platform_parameter_value"
+
+private const val INTEGER_PLATFORM_PARAMETER_NAME = "integer_platform_parameter_name"
+private const val INTEGER_PLATFORM_PARAMETER_VALUE = 1
+
+private const val BOOLEAN_PLATFORM_PARAMETER_NAME = "boolean_platform_parameter_name"
+private const val BOOLEAN_PLATFORM_PARAMETER_VALUE = true
+
+private const val INCORRECT_PLATFORM_PARAMETER_NAME = "incorrect_platform_parameter_name"
+
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = PlatformParameterSingletonTest.TestApplication::class)
+class PlatformParameterSingletonTest {
+
+  @Inject
+  lateinit var platformParameterSingleton: PlatformParameterSingleton
+
+  private val mockPlatformParameterMap by lazy {
+    val stringPlatformParameter = ParameterValue.getDefaultInstance().toBuilder()
+      .setString(STRING_PLATFORM_PARAMETER_VALUE).build()
```

Comment:
Can you not do `ParameterValue.newBuilder().setString(...).build()`? That should create the default instance anyway (same comment below)

Reply:
Yeah this is better. Done